internal/clearingway/handlers.go

```diff
@@ -466,6 +517,79 @@ func (c *Clearingway) RemoveAll(s *discordgo.Session, i *discordgo.InteractionCr
 	}
 }
 
+func (c *Clearingway) RequestReclear(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	g, ok := c.Guilds.Guilds[i.GuildID]
+	if !ok {
+		fmt.Printf("Interaction received from guild %s with no configuration!\n", i.GuildID)
+		return
+	}
+
+	// Ignore messages not on the correct channel
+	if i.ChannelID != g.ChannelId {
+		fmt.Printf("Ignoring message not in channel %s.\n", g.ChannelId)
+	}
+
+	err := discord.StartInteraction(s, i.Interaction, "Checking for respective clear role...")
+	if err != nil {
+		fmt.Printf("Error sending Discord message: %v\n", err)
+		return
+	}
+
+	ultimate := i.ApplicationCommandData().Options[0].StringValue()
+	encounter := g.Encounters.ForName(ultimate)
+	reclearRole := encounter.Roles[ReclearRole]
+	clearedRole := encounter.Roles[ClearedRole]
+
+	var rolePresent = false
+	var clearPresent = false
+	for _, r := range i.Member.Roles {
+		if r == reclearRole.DiscordRole.ID {
+			rolePresent = true
+			continue
+		}
+		if r == clearedRole.DiscordRole.ID {
+			clearPresent = true
+			continue
+		}
+	}
+
+	// Remove role no matter what
+	// Add role only if cleared role is present
+	if rolePresent {
+		fmt.Printf("Removing role: %+v\n", reclearRole.Name)
+		err = reclearRole.RemoveFromCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+		if err != nil {
+			fmt.Printf("Error removing role: %+v\n", err)
+			return
+		}
+		tempstr := fmt.Sprintf("Successfully removed role: <@&%v>", reclearRole.DiscordRole.ID)
+		err = discord.ContinueInteraction(s, i.Interaction, tempstr)
+		if err != nil {
+			fmt.Printf("Error sending Discord message: %v\n", err)
+		}
+	} else {
+		if clearPresent {
+			fmt.Printf("Adding role: %+v\n", reclearRole.Name)
+			err = reclearRole.AddToCharacter(g.Id, i.Member.User.ID, c.Discord.Session)
+			if err != nil {
+				fmt.Printf("Error adding role: %+v\n", err)
+				return
+			}
+			tempstr := fmt.Sprintf("Successfully added role: <@&%v>", reclearRole.DiscordRole.ID)
+			err = discord.ContinueInteraction(s, i.Interaction, tempstr)
+			if err != nil {
+				fmt.Printf("Error sending Discord message: %v\n", err)
+			}
+		} else {
+			tempstr := fmt.Sprintf("You do not have the required role: <@&%v>", clearedRole.DiscordRole.ID)
+			err = discord.ContinueInteraction(s, i.Interaction, tempstr)
+			if err != nil {
+				fmt.Printf("Error sending Discord message: %v\n", err)
```

Comment:
likewise to above, we could remove the role here

Reply:
Same as above